### ISSUE TITLE
Org Admins cannot enforce Private setting for a Dataset against Editor members

### ISSUE THREAD

**Issue by reedv** [ckan/ckan#5889]
**CKAN version**
2.9.1

**Describe the bug**
Editors are able to set Datasets to Public even if an Admin has set them to Private.
This conflicts with my understanding on how CKAN works as described here (https://docs.ckan.org/en/2.9/user-guide.html#users-organizations-and-authorization) and here (https://ckan.org/portfolio/publish-and-manage-data/).

**Steps to reproduce**
1. Add a user1 to Org as an Admin
2. As user1, add a Dataset and have it set to Private
3. Add a user2 to same Org as a Editor (since Members don't evenhave the option to edit the Datasets at all)
4. As user2, edit the description and set the Dataset to Public

**Expected behavior**
The Editor user2 should not be able to set the Dataset to Public that an Admin had made Private (and something should notify the Admin user1)

**Additional details**
If possible, please provide the full stack trace of the error raised,  or add screenshots to help explain your problem.

As per https://docs.ckan.org/en/2.9/user-guide.html#users-organizations-and-authorization
> An organization’s administrators can add individual users to it, with different roles depending on the level of authorization needed. A user in an organization can create a dataset owned by that organization. In the default setup, this dataset is initially private, and visible only to other users in the same organization. **When it is ready for publication, it can be published at the press of a button. This may require a higher authorization level within the organization**.

Is there some config I need to change? Am I misinterpreting the feature descriptions in the CKAN docs and site? 
Simply setting a Dataset to Private via the Bulk Edit tool as an Admin does not help either. An Editor user can still just set it back to Public without any issue.
**Basically, is there a way to have only Admins control whether a Dataset is Public/Private?**

**Comment by reedv** [comment]
This issue appears to have been brought up here (https://lists-archive.okfn.org/pipermail/ckan-dev/2014-June/007641.html) as well.

**Comment by reedv** [comment]
One workaround I can think of is to modify the file(s) that present the Edit UI/form to Editor users to remove the Private/Public switch. Could you point to where on in the CKAN files I can find this logic?

**Comment by reedv** [comment]
In the process of addressing the API question here: https://groups.google.com/a/ckan.org/g/ckan-dev/c/UKZ9iWTugvI/m/aHzgtWwlCQAJ

**Comment by reedv** [comment]
Made my own extension to do what I needed: https://github.com/reedv/ckanext-publicrestrictiondatasets

Adds restrictions to datasets so that only sysadmin users can set visibility to Public. This is in order to make CKAN's behavior match closer how it is described here (https://ckan.org/portfolio/publish-and-manage-data/).

> Admins can approve datasets for publication with our bulk editing tool which let’s you search, facets and pick datasets to become public or private.

(I still hide the API as described here https://groups.google.com/a/ckan.org/g/ckan-dev/c/UKZ9iWTugvI but now even if someone has API access they should not be able to set packages Public without being a sysadmin).

**Comment by ThrawnCA** [comment]
> This conflicts with my understanding on how CKAN works as described here (https://docs.ckan.org/en/2.9/user-guide.html#users-organizations-and-authorization) and here (https://ckan.org/portfolio/publish-and-manage-data/).

I think that that is a misreading.

The wording is, "In the default setup, this dataset is initially private, and visible only to other users in the same organization. When it is ready for publication, it can be published at the press of a button. This may require a higher authorization level within the organization."

That is to say, _viewing_ a private dataset requires only that you be "in the same organization", ie a Member, but _publishing_ it requires more than that, ie Editor.

It does not mean that publication requires higher authorization than creation. The only privilege of Admin above Editor is its ability to assign and revoke the three roles within the organization.
